### PR TITLE
Add hooks_dir parameter for custom git hooks

### PR DESCRIPTION
**Pull request by WyriHaximus**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | yes
| Fixed tickets | none

This pull request adds the `hooks_dir` parameter which allows users to overwrite the default git hooks as explaining in this upcoming blog post: https://github.com/WyriHaximus/blog.wyrihaximus.net/pull/6

### CODE REVIEW

File: src/GrumPHP/Console/Command/Git/InitCommand.php [hunk 1]
```diff
@@ -92,6 +93,9 @@ public function execute(InputInterface $input, OutputInterface $output)
         foreach (self::$hooks as $hook) {
             $gitHook = $gitHooksPath . $hook;
             $hookTemplate = $resourceHooksPath . $hook;
+            if (file_exists($customHooksPath . $hook)) {
```
Comment (veewee) [inline src/GrumPHP/Console/Command/Git/InitCommand.php]: Can you use `$this->filesystem->exists` here?

**Comment by veewee** [comment]
Thanks for your PR. It looks good! I've added one little remark in the code.
Besides of the remark, I am thinking about 2 small improvements:

- Maybe it's possible to split up the current hooks folder into multiple folders with presets like: `local`, `vagrant`, ... This way the users have a way to easily configure grumphp to run in vagrant without writing their own hook templates; This might require some additional parameters in the template file.

- It is possible to forget the trailing slash in the configurable option. Maybe it's better to make both possible? For example: `config/hooks` and 'config/hooks/'?

Thanks!

**Comment by WyriHaximus** [comment]
I've fixed the `file_exists` and trailing slash remarks. The preset folders, do you want that in this PR, or do you want me to file another? Personally doing it in this makes the most sense, I'll add the ones from my blog post.

File: src/GrumPHP/Console/Command/Git/InitCommand.php [hunk 2]
```diff
@@ -79,7 +79,7 @@ public function execute(InputInterface $input, OutputInterface $output)
         $this->input = $input;
         $gitHooksPath = $this->paths()->getGitHooksDir();
         $resourceHooksPath = $this->paths()->getGitHookTemplatesDir();
-        $customHooksPath = $this->grumPHP->getHooksDir();
+        $customHooksPath = rtrim($this->grumPHP->getHooksDir(), '/') . '/';
```
Comment (veewee) [inline src/GrumPHP/Console/Command/Git/InitCommand.php]: You probably need DIRECTORY_SEPARATOR for Windows compatibility.

**Comment by veewee** [comment]
You can add the vagrant hooks in this PR and add a this in the documentation so that the users are aware that the vagrant hooks exist.

**Comment by WyriHaximus** [comment]
Made the changes. About the vagrant hooks, I copied the ones over we use now. But would it be more desirable to use this instead? (Need to test them first though.)

```bash
#!/bin/sh
vagrant ssh --command "(cd "${HOOK_EXEC_PATH}" && exec $(HOOK_COMMAND) '--skip-success-output')"
```

**Comment by veewee** [comment]
Indeed, the location should be somehow configurable.
If it is to hard or you need vagrant specific configuration for that, it is also possible to create a grumphp-vagrant extension or something like that.

**Comment by WyriHaximus** [comment]
As long as you configure `grumphp.yml` correctly it shouldn't matter and work for the 80%. If users need anything special they can copy them and write their own using the `hooks_dir` parameter

File: src/GrumPHP/Console/Helper/PathsHelper.php [hunk 3]
```diff
@@ -206,8 +204,7 @@ public function getBinCommand($command, $forceUnix = false)
      */
     public function getRelativePath($path)
     {
-        $realpath = $this->getAbsolutePath($path);
-        return $this->fileSystem->makePathRelative($realpath, $this->getWorkingDir());
+        return $this->getAbsolutePath($path);
```
Comment (WyriHaximus) [inline src/GrumPHP/Console/Helper/PathsHelper.php]: Kinda makes this method obsolete, will refactor it out if absolute paths are the way forward

**Comment by WyriHaximus** [comment]
Reverted the path change and added a bit of docs for when using vagrant. Not sure if it is sufficient though

**Comment by veewee** [comment]
Looks good! I'll add some additional documentation.

**Comment by WyriHaximus** [comment]
Awesome, thanks :+1:

**Pull request merged**